### PR TITLE
CMP: Fix total_timeout behavior; small doc and diagnostic improvements

### PR DESCRIPTION
**Pull request by DDvO**
Also fix up CSR test cases for `81-test_cmp_cli.t`

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by DDvO** [comment]
Rebased to fix trivial merge conflict (due to renamed files changed here).

**Comment by DDvO** [comment]
Rebased again to fix a merge conflict.

File: crypto/cmp/cmp_client.c [hunk 1]
```diff
@@ -712,23 +725,8 @@ X509 *OSSL_CMP_exec_certreq(OSSL_CMP_CTX *ctx, int req_type,
         ERR_raise(ERR_LIB_CMP, CMP_R_NULL_ARGUMENT);
         return NULL;
     }
-    if (is_p10 && crm != NULL) {
-        ERR_raise(ERR_LIB_CMP, CMP_R_INVALID_ARGS);
-        return NULL;
-    }
```
Comment (t8m) [inline crypto/cmp/cmp_client.c:718]: Is this check not needed?

**Review by t8m — APPROVED** [review]
Approval pending on the question

File: crypto/cmp/cmp_client.c (hunk 1 above)
Comment (DDvO) [reply crypto/cmp/cmp_client.c:718]: This check has moved to `cmp_msg.c`:
```
    if (type == OSSL_CMP_PKIBODY_P10CR && crm != NULL) {
        ERR_raise(ERR_LIB_CMP, CMP_R_INVALID_ARGS);
        return NULL;
    }
```

Comment (t8m) [reply crypto/cmp/cmp_client.c:718]: I've seen that but it was not obvious to me whether it is truly equivalent as it is done at a very different place. OK, anyway.

Comment (DDvO) [reply crypto/cmp/cmp_client.c:718]: Good anyway that you questioned this.
`OSSL_CMP_exec_certreq()` calls `initial_certreq()` basically right away,
which calls `ossl_cmp_certreq_new()`, which does the same check
(where `is_p10 = req_type == OSSL_CMP_PKIBODY_P10CR`).

**Comment by DDvO** [comment]
Merged - thanks @t8m

**Pull request closed without merging**